---
title: "pacific_herring"
author: "laura e timm"
date: '2024-01-25'
output: html_document
---

# setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## load libraries
```{r load_libraries, warning=FALSE, message=FALSE}

#install.packages("rgdal", type="source")
#install.packages("remotes")
#remotes::install_github("royfrancis/pophelper")

packages_needed <- c("tidyverse", "ggplot2", "readxl",
                     "ggOceanMaps", "pophelper", "stringr",
                     "ggspatial", "gridExtra", "ggpubr",
                     "gtools", "reshape2", "readr")

for(i in 1:length(packages_needed)){
  if(!(packages_needed[i] %in% installed.packages())){install.packages(packages_needed[i])}
  library(packages_needed[i], character.only = TRUE)
}

```


## define functions

### plot pca
```{r plot_pca_fx}

plot_pca <- function(FEATURE_DF, COV_MATRIX) {
  pca_e <- eigen(COV_MATRIX)
  first3pcs <- data.frame(pca_e$vectors[,1:3])
  labeled_first3pcs <- cbind(FEATURE_DF,
                             pc1 = first3pcs$X1,
                             pc2 = first3pcs$X2,
                             pc3 = first3pcs$X3)
  pca_graph <- ggplot(labeled_first3pcs, aes(x = pc1, y = pc2, color = feature, shape = spawning)) + 
    geom_point(size = 3) +
    scale_color_manual(values=opt1_region_palette) +
    xlab(paste0("PC1 - ", sprintf("%0.2f", pca_e$values[1]), "%")) +
    ylab(paste0("PC2 - ", sprintf("%0.2f", pca_e$values[2]), "%"))
}

```

### find optimal k for admixture analysis
```{r optimal_k_fx}

opt_k_fx <- function(LOG_FILENAMES, K_VAL) {
  bigData<-lapply(1:length(LOG_FILENAMES), FUN = function(i) readLines(LOG_FILENAMES[i]))
  foundset<-sapply(1:length(LOG_FILENAMES), FUN= function(x) bigData[[x]][which(str_sub(bigData[[x]], 1, 1) == 'b')])
  as.numeric( sub("\\D*(\\d+).*", "\\1", foundset) )
  logs<-data.frame(K = rep(1:K_VAL, each=3))
  logs$like<-as.vector(as.numeric( sub("\\D*(\\d+).*", "\\1", foundset) ))
  tapply(logs$like, logs$K, FUN= function(x) mean(abs(x))/sd(abs(x)))
}
  
```

### plot admixture
```{r define_admixplot_fx}

admix_plot <- function(FEAT_DF, qopt_files, OUT_DIR, OUT_NAME) {
  groupings <- data.frame(FEAT_DF$ordered_region)
  admix_list <- readQ(qopt_files)
  k_list <- c()
  for (i in 2:length(admix_list)) {
    k_list <- c(k_list, paste0("k = ", as.character(i)))
  }
  
  p1 <- plotQ(alignK(admix_list[2:4]),
            grplab = groupings,
            #selgrp = "region",
            #subsetgrp = ordered_regions,
            grplabangle = 45,
            grplabpos = 0.6,
            grplabheight = 2,
            grplabjust = 1,
            panelratio = c(2,2),
            grplabsize = 8,
            splab = k_list[1:3],
            splabsize =18,
            ordergrp = T,
            clustercol = c("#416044","#31395a","#b0c7a2","#7580aa"),
            #showdiv = T,
            #divgrp = "region",
            imgoutput = "join",
            returnplot = T,
            exportplot = T,
            basesize = 11,
            sortind = "Cluster1",
            sharedindlab = F,
            imgtype = "jpeg",
            width = 20,
            height = 4,
            units = "in",
            dpi = 600,
            exportpath = OUT_DIR,
            outputfilename = OUT_NAME)
  grid.arrange(p1$plot[[1]])
}

```

### get ld df
```{r define_get_ld_df_fx}

get_LD_df <- function(FILENAME){
  r <- read.table(FILENAME, header = FALSE, stringsAsFactors = FALSE)
  
  colnames(r) <- c("snp1", "snp2", "bp_space", "r2", "D", "Dp", "r2_EM")
  
  r_ord <- r[order(r$snp1, r$snp2),]
  for(i in 3:ncol(r_ord)){r_ord[,i] <- as.numeric(r_ord[,i])}
  for(j in 1:2){r_ord[,j] <- as.character(r_ord[,j])}
  
  r_ord
  
}

```

### plot ld heatmap
```{r define_plot_ld_heatmap_fx}

plot_ld_heatmap <- function(LD_DF){
  LDHM <- ggplot(data = LD_DF, aes(snp1, snp2, fill = r2))+
    geom_tile()+
	  scale_fill_gradient2(low = "white", 
	                       high = "navyblue", 
	                       mid = "turquoise",
	                       midpoint = 0.5,
	                       limit = c(0,1),
	                       space = "Lab", 
	                       name="LD") +
    ggtitle("linkage disequilibrium")+
    theme_minimal()+ 
    theme(axis.text.x=element_blank(),
          axis.ticks.x=element_blank(), 
          axis.text.y=element_blank(),  
	  		  axis.ticks.y=element_blank())+
  			  coord_fixed()+
   			  theme(panel.grid.major = element_blank(),
   			        panel.grid.minor = element_blank(),
   			        panel.background = element_blank(), 
   			        axis.line = element_line(colour = "black"),
   			        legend.key.size = unit(1, 'cm'))
  
  LDHM
  
}

```

## specify
```{r project_setup}

BASEDIR <- "C:/Users/laura.timm/Desktop/Clupea_pallasii120/" #the directory where the metadata file lives
FIG_DIR <- paste0(BASEDIR, "FIGURES/")
PREFIX <- "CPAL-CHARFULL" #the prefix for the lcWGS run
METADATAFILE <- paste0(BASEDIR, "pacific-herring_metadata_120.xlsx") #metadatafile name (excel file)
SHEET <- "Sheet1" #sheet containing the pertinent metadata
DEPTH_LOWER_BOUND <- 1

ADMIX_PREFIX_all <- "admixture/CPAL-CHARFULL_wgph"

CHROMFILE <- paste0(BASEDIR, "CHAR_chrom_metadata.txt")
chrom_df <- read.table(CHROMFILE, header = TRUE)

metadata <- read_xlsx(METADATAFILE, sheet = SHEET)

#teal and orange
opt1_region_palette <- c("Togiak" = "#2a5674",
                         "Port Moller" = "#4f90a6",
                         "Constantine Bay" = "#85c4c9",
                         "Popof Island" = "#fdd0a2",
                         "Kodiak - Kiliuda" = "#fd8d3c",
                         "Kodiak - Uganik" = "#d94801",
                         "Cordova" = "#8c2d04",
                         "A" = "#51087E",
                         "B" = "#880ED4", 
                         "C" = "#C576F6")

```

```{r additional_setup}

#hierarchical admixture
ADMIX_PREFIX_EBS <- "admixture/CPAL-CHARFULL-eBS_wgph"
metadata_ebs <- metadata %>%
  filter(twopops_nuc_cluster == "eBSAI")

ADMIX_PREFIX_GOA <- "admixture/CPAL-CHARFULL-GOA_wgph"
metadata_goa <- metadata %>%
  filter(twopops_nuc_cluster == "nGOA")

#eBSAIclust
#genotype heatmap
ABC_BEAGLE_FILENAME <- paste0(BASEDIR, "genotype_heatmap/CPAL-CHARFULL-eBS_chr7region_wgph.beagle")
AC_BEAGLE_FILENAME <- paste0(BASEDIR, "genotype_heatmap/CPAL-CHARFULL-eBS-AC_chr7region.beagle")

```

# all samples (eBS + GOA)

## Mapping collection locations

```{r map_samples}

specimen_samples <- count(metadata, Latitude_DD, Longitude_DD, MarineRegion, spawning)

(m <- basemap(limits = c(-167, -145, 53, 61), bathymetry = T, rotate = T) +
    geom_spatial_point(data = specimen_samples,
                       size = 3,
                       aes(x = Longitude_DD, 
                           y = Latitude_DD,
                           shape = spawning,
                           color = MarineRegion)) +
    scale_color_manual(values=opt1_region_palette) +
    annotation_scale() + #scale bar
    annotation_north_arrow(location = "tr", which_north = "true", height = unit(0.75,"cm"),width = unit(0.75,"cm"))) #north arrow

#ggsave(paste0(FIG_DIR, PREFIX, "color-opt1_size3_mapFINAL.pdf"), width = 6, height = 7, units = "in", dpi = 600)

```



## Aligned sequencing depths

```{r plot_initial_depth_distributions, warning=FALSE, message=FALSE}

just_depths <- read.csv(paste0(BASEDIR, PREFIX, "_depths.csv"), header = FALSE, sep = '\t')
colnames(just_depths) <- c("Barcode", "mean_depth")
full_metadata <- left_join(metadata, just_depths, by = "Barcode")

theme_set(
  theme( 
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, size = 10, vjust = 0.5),
    axis.text.y = element_text(angle = 0, size = 10),
    panel.background = element_rect(fill = "white"), 
    panel.spacing = unit(0,"lines"),
    strip.text.y = element_text(angle = 0) 
  )
)

(depths_dists_plot <- ggplot(full_metadata, aes(x = mean_depth, color = MarineRegion)) +
    geom_density(n = 50, linewidth = 1) +
    geom_vline(aes(xintercept = DEPTH_LOWER_BOUND), colour = "black") +
    scale_color_manual(values=opt1_region_palette) +
    ggtitle("individual mean sequence depth distributions"))

#ggsave(paste0(BASEDIR, PREFIX, "_region_mean_depths.jpg"), width = 8, height = 5, units = "in", dpi = 300)

```

## PCA

```{r pca_all}

covfile_all <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "_wgph.cov")))
features_df_all <- metadata[c("Barcode", "MarineRegion", "spawning")]
colnames(features_df_all) <- c("Barcode", "feature", "spawning")

(pca_all <- plot_pca(features_df_all, covfile_all))

#ggsave(paste0(FIG_DIR, PREFIX, "_PCA_color-opt1_size3.jpg"), width = 6, height = 4, units = "in", dpi = 600)

```


### PCA by region
```{r pca_grid_spawn_region, fig.width = 8, fig.heigth = 20}

pca_e_facet <- eigen(covfile_all)
first3pcs_facet <- data.frame(pca_e_facet$vectors[,1:3])
labeled_first3pcs_facet <- cbind(metadata,
                           pc1 = first3pcs_facet$X1,
                           pc2 = first3pcs_facet$X2,
                           pc3 = first3pcs_facet$X3)
(pca_facet <- ggplot(labeled_first3pcs_facet, aes(x = pc1, y = pc2, color = MarineRegion, shape = spawning)) + 
  geom_point(size = 3, alpha = 0.75) +
  scale_color_manual(values=opt1_region_palette) +
  xlab(paste0("PC1 - ", sprintf("%0.2f", pca_e_facet$values[1]), "%")) +
  ylab(paste0("PC2 - ", sprintf("%0.2f", pca_e_facet$values[2]), "%")) +
  facet_grid(rows = vars(MarineRegion)))

#ggsave(paste0(FIG_DIR, PREFIX, "_PCA_region_facet_color-opt1_size3.jpg"), width = 8, height = 12, units = "in", dpi = 600)

```



## Admixture
```{r opt_k_all}

LOGS_all <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_all, "*.log"))
opt_k_fx(LOGS_all, 7)

```



```{r admix_plot_all}

QOPTS_all <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_all, "_*-2.qopt"))
#oname <- paste0(PREFIX, "_K23-admix_by-clust_TOG2COR")
#features_df_admix <- metadata[c("Barcode", "admix_grp_order_TOG2COR")]
oname <- paste0(PREFIX, "_K23-admix_by-clust_COR2TOG")
features_df_admix <- metadata[c("Barcode", "admix_grp_order_COR2TOG")]
colnames(features_df_admix) <- c("Barcode", "ordered_region")
(all_samples_admix_plot <- admix_plot(features_df_admix, QOPTS_all, FIG_DIR, oname))

```



Let's start considering these two clusters (eBS and GOA as separate groups):
"eBS" = eBS, Constantine Bay, and Port Moller
"GOA" = GOA, Kodiak, and Popof

# eBS

## PCA

### N = 60
```{r pca_eBS}

eBS_metadata <- metadata %>%
  filter(twopops_nuc_cluster == "eBSAI")
features_df_eBS <- eBS_metadata[c("Barcode", "MarineRegion", "spawning")]
colnames(features_df_eBS) <- c("Barcode", "feature", "spawning")

covfile_eBS <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-eBS_wgph.cov")))

(pca_eBS <- plot_pca(features_df_eBS, covfile_eBS))

#ggsave(paste0(BASEDIR, PREFIX, "-eBS_PCA_region.jpg"), width = 6, height = 4, units = "in", dpi = 600)

```



Maybe two or three groups? Remove outliers to see it better.

### N = 59

```{r pca_eBS_59}

#identify which individuals to filter out
pca_e_eBS <- eigen(covfile_eBS)
first3pcs_eBS <- data.frame(pca_e_eBS$vectors[,1:3])
labeled_first3pcs_eBS <- cbind(features_df_eBS,
                             pc1 = first3pcs_eBS$X1,
                             pc2 = first3pcs_eBS$X2,
                             pc3 = first3pcs_eBS$X3)
(filter_eBS_59 <- labeled_first3pcs_eBS[labeled_first3pcs_eBS$pc1 > 0.6,])

eBS_metadata_59 <- labeled_first3pcs_eBS[labeled_first3pcs_eBS$pc1 < 0.6,]
features_df_eBS_59 <- eBS_metadata_59[c("Barcode", "feature", "spawning")]

covfile_eBS_59 <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-eBS_wgph_keep59.cov")))

(pca_eBS_59 <- plot_pca(features_df_eBS_59, covfile_eBS_59))

#ggsave(paste0(BASEDIR, PREFIX, "-eBS59_PCA_region_color-opt1.jpg"), width = 5.5, height = 4, units = "in", dpi = 600)

```


Let's assign individuals to clusters for manhattan plots!
```{r eBS59_classification}

pca_e_eBS59 <- eigen(covfile_eBS_59)
first3pcs_eBS59 <- data.frame(pca_e_eBS59$vectors[,1:3])
labeled_first3pcs_eBS59 <- cbind(features_df_eBS_59,
                             pc1 = first3pcs_eBS59$X1,
                             pc2 = first3pcs_eBS59$X2,
                             pc3 = first3pcs_eBS59$X3)

#upper lefthand cluster
eBS59A <- labeled_first3pcs_eBS59 %>%
  filter(pc1 < 0.0 & pc2 > 0.0)
#lower lefthand cluster
eBS59B <- labeled_first3pcs_eBS59 %>%
  filter(pc1 < 0.0 & pc2 < 0.0)
#righthand cluster
eBS59C <- labeled_first3pcs_eBS59 %>%
  filter(pc1 > 0.15 & pc1 <0.3 & pc2 < 0.1)

```

Yes, structure, but NOT related to geography of spawning pops (Togiak and Port Moller).

## Admixture
```{r opt_k_ebs}

LOGS_EBS <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_EBS, "*.log"))
opt_k_fx(LOGS_EBS, 7)

```



```{r admix_plot_ebs}

QOPTS_ebs <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_EBS, "_*-2.qopt"))
oname_ebs <- paste0(PREFIX, "-eBS_K24-admix_by-clust")
features_df_admix_ebs <- metadata_ebs[c("Barcode", "admix_grp_order_COR2TOG")]
colnames(features_df_admix_ebs) <- c("Barcode", "ordered_region")
(ebs_samples_admix_plot <- admix_plot(features_df_admix_ebs, QOPTS_ebs, FIG_DIR, oname_ebs))

```


## eBSAI nuclear clusters: chr7 region
targeting the region on chr7 (NC-045158.1:13841600-13851100)

### three clusters (A, B, C)
#### PCA of region
```{r eBSAI_chr7region_pca}

covfile_ebs_chr7region <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-eBS_chr7region_wgph.cov")))
features_df_ebs <- metadata_ebs[c("Barcode", "eBSAI_clust")]

pca_e_ebs_chr7region <- eigen(covfile_ebs_chr7region)
first3pcs_ebs_chr7region <- data.frame(pca_e_ebs_chr7region$vectors[,1:3])
labeled_first3pcs_ebs_chr7region <- cbind(features_df_ebs,
                                          pc1 = first3pcs_ebs_chr7region$X1,
                                          pc2 = first3pcs_ebs_chr7region$X2,
                                          pc3 = first3pcs_ebs_chr7region$X3)
(pca_graph_ebs_chr7region <- ggplot(labeled_first3pcs_ebs_chr7region, 
                                   aes(x = pc1, y = pc2, color = eBSAI_clust)) + 
  geom_point(size = 3) +
  scale_color_manual(values=opt1_region_palette) +
  xlab(paste0("PC1 - ", sprintf("%0.2f", pca_e_ebs_chr7region$values[1]), "%")) +
  ylab(paste0("PC2 - ", sprintf("%0.2f", pca_e_ebs_chr7region$values[2]), "%")))

ggsave(paste0(FIG_DIR, PREFIX, "-eBS_chr7region_PCA_color-opt1_size3.jpg"), width = 6, height = 4, units = "in", dpi = 600)

```



#### genotype heatmap of region
```{r ebs_chr7region_genoheatmap}

ebs_beagle_file <- read.table(ABC_BEAGLE_FILENAME)
ebs_geno_matrix <- matrix(ncol = ((ncol(ebs_beagle_file)-3)/3), nrow = (nrow(ebs_beagle_file)-1))
colnames(ebs_geno_matrix) <- paste0("Ind",1:((ncol(ebs_beagle_file)-3)/3))
for(i in 2:nrow(ebs_beagle_file)){
  for(j in 6:ncol(ebs_beagle_file)){
    if(j%%3 == 0){
      hom_major <- as.double(ebs_beagle_file[i,(j-2)])
      het <- as.double(ebs_beagle_file[i,(j-1)])
      hom_minor <- as.double(ebs_beagle_file[i,(j)])
      probs <- c(hom_major, het, hom_minor)
      
      if(probs[1] == probs[2] & probs[1] == probs[3]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- NA
      } else if(probs[1] == max(probs) & probs[1] == probs[2]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- 1 + probs[1]
      } else if(probs[1] == max(probs) & probs[1] == probs[3]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- "NO"
      } else if(probs[2] == max(probs) & probs[2] == probs[3]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- 3 + probs[2]
      } else if(max(probs) == probs[1]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- abs(probs[1]-1)
      } else if(max(probs) == probs[2]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- 2 + probs[2]
      } else if(max(probs) == probs[3]){
        ebs_geno_matrix[(i-1),((j/3)-1)] <- 4 + probs[3]
      }
    }
  } 
}
  
## convert matrix to datatable for plotting
ebs_geno_table <- reshape2::melt(ebs_geno_matrix)
  
## Rename columns 
colnames(ebs_geno_table) <- c("locus", "Ind", "value")

ind_list <- c()
for (ind_num in 1:nrow(features_df_ebs)) {
  ind_list <- append(ind_list, paste0("Ind", ind_num))
}

features_df_ebs$Ind <- ind_list
ebs_geno_plot <- left_join(ebs_geno_table, features_df_ebs)

## Factor the feature into the order you want them to plot
ebs_geno_plot$cluster <- factor(ebs_geno_plot$eBSAI_clust, levels = c("A", "B", "C"))

#order for plotting
ebs_geno_plot$Ind <- factor(ebs_geno_plot$Ind, 
                            levels = unique((ebs_geno_plot$Ind)[order(ebs_geno_plot$cluster)]))

# Plot your genotype heatmap 
ebs_geno_heatmap <- ggplot(ebs_geno_plot, aes(x = locus, y = Ind, fill = value)) +
  geom_tile() +
  theme_minimal()+ 
  theme(axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(), 
        axis.line = element_line(colour = "black"), 
        text = element_text(size = 12), 
        legend.key.size = unit(1, 'cm')) +
  ylab(label ="Individual")+
  scale_fill_gradient2(low = "blue", 
                       mid = "yellow", 
                       high ="red",
                       midpoint = 3.0, 
                       guide = "colourbar", 
                       na.value = "white")
## Make the colored bar that is for your population colors 
## add a column called Locality_text that just has Locality as all the values 
## this will serve as your X-axis and your population ID will serve as your y-axis
ebs_geno_plot <- ebs_geno_plot %>% 
  mutate(cluster_text = "cluster")

# plot a bar that represents population ids for the individuals in rows of the genotype heatmap
ebs_geno_bar <- ggplot(ebs_geno_plot, aes(cluster_text, Ind, fill = cluster)) +
  geom_tile() + 
  theme_minimal()+ 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),  
        axis.line = element_blank(), 
        legend.text = element_text(size = 12), 
        legend.key.size = unit(1, 'cm')) +
  theme(axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  xlab(label = ". ") +
  ylab(label = " ") +
  scale_fill_manual(values = opt1_region_palette) + 
  theme(legend.position="left") + 
  theme(axis.text.x=element_blank())

(ebs_full_geno_heatmap <- ggarrange(ebs_geno_bar, ebs_geno_heatmap, widths = c(0.3, 1)))

ggsave(paste0(FIG_DIR, PREFIX, "-eBS_chr7region_genotype-heatmap.jpg"), width = 12, height = 8, units = "in", dpi = 600)

```


#### LD heatmap
```{r plot_LD}

ld_filename <- paste0(BASEDIR, "linkage/", PREFIX, "-eBS_chr7region.ld")
ld_df <- get_LD_df(ld_filename)
(ld_heatmap <- plot_ld_heatmap(ld_df))
ggsave(paste0(FIG_DIR, PREFIX, "-eBS_chr7region", "_r2.jpeg"),
       width = 12, height = 12,
       dpi = 600, units = "in")
  
```



### two clusters (A and C)
#### PCA of region
```{r eBSAI_AC_chr7region_pca}

covfile_ebs_AC_chr7region <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-eBS-AC_chr7region_wgph.cov")))
features_df_ebs_AC <- features_df_ebs %>%
  filter(eBSAI_clust == "A" | eBSAI_clust == "C")

pca_e_ebs_AC_chr7region <- eigen(covfile_ebs_AC_chr7region)
first3pcs_ebs_AC_chr7region <- data.frame(pca_e_ebs_AC_chr7region$vectors[,1:3])
labeled_first3pcs_ebs_AC_chr7region <- cbind(features_df_ebs_AC,
                                          pc1 = first3pcs_ebs_AC_chr7region$X1,
                                          pc2 = first3pcs_ebs_AC_chr7region$X2,
                                          pc3 = first3pcs_ebs_AC_chr7region$X3)
(pca_graph_ebs_chr7region <- ggplot(labeled_first3pcs_ebs_AC_chr7region, 
                                   aes(x = pc1, y = pc2, color = eBSAI_clust)) + 
  geom_point(size = 3) +
  scale_color_manual(values=opt1_region_palette) +
  xlab(paste0("PC1 - ", sprintf("%0.2f", pca_e_ebs_AC_chr7region$values[1]), "%")) +
  ylab(paste0("PC2 - ", sprintf("%0.2f", pca_e_ebs_AC_chr7region$values[2]), "%")))

ggsave(paste0(FIG_DIR, PREFIX, "-eBS-AC_chr7region_PCA_color-opt1_size3.jpg"), width = 6, height = 4, units = "in", dpi = 600)

```



#### genotype heatmap of region
```{r ebs_AC_chr7region_genoheatmap}

ebs_AC_beagle_file <- read.table(AC_BEAGLE_FILENAME)
ebs_AC_geno_matrix <- matrix(ncol = ((ncol(ebs_AC_beagle_file)-3)/3), nrow = (nrow(ebs_AC_beagle_file)-1))
colnames(ebs_AC_geno_matrix) <- paste0("Ind",1:((ncol(ebs_AC_beagle_file)-3)/3))
for(i in 2:nrow(ebs_AC_beagle_file)){
  for(j in 6:ncol(ebs_AC_beagle_file)){
    if(j%%3 == 0){
      hom_major <- as.double(ebs_AC_beagle_file[i,(j-2)])
      het <- as.double(ebs_AC_beagle_file[i,(j-1)])
      hom_minor <- as.double(ebs_AC_beagle_file[i,(j)])
      probs <- c(hom_major, het, hom_minor)
      
      if(probs[1] == probs[2] & probs[1] == probs[3]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- NA
      } else if(probs[1] == max(probs) & probs[1] == probs[2]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- 1 + probs[1]
      } else if(probs[1] == max(probs) & probs[1] == probs[3]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- "NO"
      } else if(probs[2] == max(probs) & probs[2] == probs[3]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- 3 + probs[2]
      } else if(max(probs) == probs[1]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- abs(probs[1]-1)
      } else if(max(probs) == probs[2]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- 2 + probs[2]
      } else if(max(probs) == probs[3]){
        ebs_AC_geno_matrix[(i-1),((j/3)-1)] <- 4 + probs[3]
      }
    }
  } 
}
  
## convert matrix to datatable for plotting
ebs_AC_geno_table <- reshape2::melt(ebs_AC_geno_matrix)
  
## Rename columns 
colnames(ebs_AC_geno_table) <- c("locus", "Ind", "value")

ind_list <- c()
for (ind_num in 1:nrow(features_df_ebs_AC)) {
  ind_list <- append(ind_list, paste0("Ind", ind_num))
}

features_df_ebs_AC$Ind <- ind_list
ebs_AC_geno_plot <- left_join(ebs_AC_geno_table, features_df_ebs_AC)

## Factor the feature into the order you want them to plot
ebs_AC_geno_plot$cluster <- factor(ebs_AC_geno_plot$eBSAI_clust, levels = c("A", "B", "C"))

#order for plotting
ebs_AC_geno_plot$Ind <- factor(ebs_AC_geno_plot$Ind, 
                            levels = unique((ebs_AC_geno_plot$Ind)[order(ebs_AC_geno_plot$cluster)]))

# Plot your genotype heatmap 
ebs_AC_geno_heatmap <- ggplot(ebs_AC_geno_plot, aes(x = locus, y = Ind, fill = value)) +
  geom_tile() +
  theme_minimal()+ 
  theme(axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank())+
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(), 
        axis.line = element_line(colour = "black"), 
        text = element_text(size = 12), 
        legend.key.size = unit(1, 'cm')) +
  ylab(label ="Individual")+
  scale_fill_gradient2(low = "blue", 
                       mid = "yellow", 
                       high ="red",
                       midpoint = 3.0, 
                       guide = "colourbar", 
                       na.value = "white")
## Make the colored bar that is for your population colors 
## add a column called Locality_text that just has Locality as all the values 
## this will serve as your X-axis and your population ID will serve as your y-axis
ebs_AC_geno_plot <- ebs_AC_geno_plot %>% 
  mutate(cluster_text = "cluster")

# plot a bar that represents population ids for the individuals in rows of the genotype heatmap
ebs_AC_geno_bar <- ggplot(ebs_AC_geno_plot, aes(cluster_text, Ind, fill = cluster)) +
  geom_tile() + 
  theme_minimal()+ 
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        panel.background = element_blank(),  
        axis.line = element_blank(), 
        legend.text = element_text(size = 12), 
        legend.key.size = unit(1, 'cm')) +
  theme(axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(),
        axis.ticks.y=element_blank()) +
  xlab(label = ". ") +
  ylab(label = " ") +
  scale_fill_manual(values = opt1_region_palette) + 
  theme(legend.position="left") + 
  theme(axis.text.x=element_blank())

(ebs_AC_full_geno_heatmap <- ggarrange(ebs_AC_geno_bar, ebs_AC_geno_heatmap, widths = c(0.3, 1)))

ggsave(paste0(FIG_DIR, PREFIX, "-eBS-AC_chr7region_genotype-heatmap.jpg"), width = 12, height = 8, units = "in", dpi = 600)

```


#### LD heatmap
```{r plot_LD_AC}

ac_ld_filename <- paste0(BASEDIR, "linkage/", PREFIX, "-eBS-AC_chr7region.ld")
ac_ld_df <- get_LD_df(ac_ld_filename)
(ac_ld_heatmap <- plot_ld_heatmap(ac_ld_df))
ggsave(paste0(FIG_DIR, PREFIX, "-eBS-AC_chr7region", "_r2.jpeg"),
       width = 12, height = 12,
       dpi = 600, units = "in")
  
```



# Gulf of Alaska

## PCA

### N = 60
```{r pca_GOA}

GOA_metadata <- metadata %>%
  filter(twopops_nuc_cluster == "nGOA")

features_df_GOA <- GOA_metadata[c("Barcode", "MarineRegion", "spawning")]

colnames(features_df_GOA) <- c("Barcode", "feature", "spawning")

covfile_GOA <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-GOA_wgph.cov")))

(pca_GOA <- plot_pca(features_df_GOA, covfile_GOA))

#ggsave(paste0(BASEDIR, PREFIX, "-GOA_PCA_region_color-op1.jpg"), width = 6, height = 4, units = "in", dpi = 600)

```

Popof Island might be disguising underlying structure. Let's remove Popof and try GOA40

### N = 40

```{r pca_GOA_noPopof}

features_df_GOA_noPopof <- features_df_GOA %>%
  filter(feature != "Popof Island")

covfile_GOA_noPopof <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-GOA-noPopof_wgph.cov")))

(pca_GOA_noPopof <- plot_pca(features_df_GOA_noPopof, covfile_GOA_noPopof))

#need to change the plot_pca fx to shape = 17 in geom_point

#ggsave(paste0(BASEDIR, PREFIX, "-GOA-noPopof_PCA_region_color-opt1.jpg"), width = 6, height = 4, units = "in", dpi = 600)

```

For the pub:
```{r pca_ggarrange}

(pca_fig <- ggarrange(pca_all, pca_eBS_59, pca_GOA, pca_GOA_noPopof, ncol = 2, nrow = 2, common.legend=TRUE))
ggsave(paste0(FIG_DIR, "PCA_FIGURE.jpg"), width = 8, height = 8, units = "in", dpi = 600)

```


Let's move forward with GOA with Popof

## Admixture
```{r opt_k_goa}

LOGS_GOA <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_GOA, "*.log"))
opt_k_fx(LOGS_GOA, 7)

```



```{r admix_plot_goa}

QOPTS_goa <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_GOA, "_*-2.qopt"))
oname_goa <- paste0(PREFIX, "-GOA_K24-admix_by-region")
features_df_admix_goa <- metadata_goa[c("Barcode", "admix_grp_order_COR2TOG")]
colnames(features_df_admix_goa) <- c("Barcode", "ordered_region")
(goa_samples_admix_plot <- admix_plot(features_df_admix_goa, QOPTS_goa, FIG_DIR, oname_goa))

```


