---
title: "pacific_herring"
author: "laura e timm"
date: '2024-01-25'
output: html_document
---

# setup
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## load libraries
```{r load_libraries, warning=FALSE, message=FALSE}

packages_needed <- c("dplyr", "tidyverse", "ggplot2",
                     "readxl", "RColorBrewer", "ggpubr",
                     "stats", "scales", "rgdal", "ggmap",
                     "ggOceanMaps", "wesanderson", "pophelper",
                     "gridExtra", "label.switching", "remotes",
                     "stringr", "writexl", "ggpmisc")

for(i in 1:length(packages_needed)){
  if(!(packages_needed[i] %in% installed.packages())){install.packages(packages_needed[i])}
  library(packages_needed[i], character.only = TRUE)
}

```


## define functions

### plot pca
```{r plot_pca_fx}

plot_pca <- function(FEATURE_DF, COV_MATRIX) {
  pca_e <- eigen(COV_MATRIX)
  first3pcs <- data.frame(pca_e$vectors[,1:3])
  labeled_first3pcs <- cbind(FEATURE_DF,
                             pc1 = first3pcs$X1,
                             pc2 = first3pcs$X2,
                             pc3 = first3pcs$X3)
  pca_graph <- ggplot(labeled_first3pcs, aes(x = pc1, y = pc2, color = feature)) + 
    geom_point(size = 2, alpha = 0.5) +
    xlab(paste0("PC1 - ", sprintf("%0.2f", pca_e$values[1]), "%")) +
    ylab(paste0("PC2 - ", sprintf("%0.2f", pca_e$values[2]), "%"))
}

```

### find optimal k for admixture analysis
```{r optimal_k_fx}

opt_k_fx <- function(LOG_FILENAMES, K_VAL) {
  bigData<-lapply(1:length(LOG_FILENAMES), FUN = function(i) readLines(LOG_FILENAMES[i]))
  foundset<-sapply(1:length(LOG_FILENAMES), FUN= function(x) bigData[[x]][which(str_sub(bigData[[x]], 1, 1) == 'b')])
  as.numeric( sub("\\D*(\\d+).*", "\\1", foundset) )
  logs<-data.frame(K = rep(1:K_VAL, each=3))
  logs$like<-as.vector(as.numeric( sub("\\D*(\\d+).*", "\\1", foundset) ))
  tapply(logs$like, logs$K, FUN= function(x) mean(abs(x))/sd(abs(x)))
}
  
```

### plot admixture
```{r define_admixplot_fx}

admix_plot <- function(FEAT_DF, qopt_files) {
  groupings <- data.frame(FEAT_DF$feature)
  colnames(groupings) <- c("feature")
  admix_list <- readQ(qopt_files)
  k_list <- c()
  for (i in 2:length(admix_list)) {
    k_list <- c(k_list, paste0("k = ", as.character(i)))
  }
  
  p1 <- plotQ(alignK(admix_list[2:length(admix_list)]),
            grplab = groupings,
            selgrp = "feature",
            grplabangle = 90,
            grplabpos = 0.6,
            grplabheight = 1,
            grplabjust = 1,
            panelratio = c(2,2),
            grplabsize = 4,
            splab = k_list,
            splabsize = 14,
            ordergrp = T,
            imgoutput = "join",
            returnplot = T,
            exportplot = F,
            basesize = 11,
            sortind = "all",
            sharedindlab = F)
  grid.arrange(p1$plot[[1]])
}

```

### manhattan plots
```{r define_manhattanplotFX}

manhattanplot <- function(CHROMS_DF, POPLIST, FSTFILE, PT_COLOR, MANHTITLE) {
  fst_df <- FSTFILE %>%
    set_names(nm = FSTFILE) %>%
    map_dfr(
      ~ read_delim(.x, 
                   skip = 1, 
                   col_types = cols(), 
                   col_names = c("region", "chr", "midpos", "nsites", "fst"), 
                   delim = "\t"),
      .id = "comparison"
      )
  fst_df <- fst_df %>%
    filter(chr %in% CHROMS_DF$chr)
  fst_df <- left_join(fst_df, CHROMS_DF, by = "chr")
  fst_df <- fst_df %>%
    mutate(midpos_Mb = midpos/1000000)
  
  final_fst_df <- fst_df %>%
    mutate(tmp1 = gsub(paste0(BASEDIR, "fst/"), "", comparison)) %>%
    mutate(tmp1 = gsub(".fst.txt", "", tmp1)) %>%
    mutate(POP1 = gsub(POPLIST[2], "", tmp1)) %>%
    mutate(POP1 = gsub("-", "", POP1)) %>%
    mutate(POP2 = gsub(POPLIST[1], "", tmp1)) %>%
    mutate(POP2 = gsub("-", "", POP2))
  final_fst_df$POP2 <- factor(final_fst_df$POP2, levels = POPLIST)
  final_fst_df$chr_num <- factor(final_fst_df$chr_num, levels = CHROMS_DF$chr_num)
  final_fst_df$fst_neg0 <- final_fst_df$fst
  final_fst_df$fst_neg0[final_fst_df$fst_neg0 < 0] <- 0
  
  #plot the full genome
  theme_set(
    theme(
      legend.position = "none",
      panel.grid.major = element_blank(),
      panel.grid.minor = element_blank(),
      axis.text.x = element_text(angle = 90, size = 10, vjust = 0.5),
      axis.text.y = element_text(angle = 0, size = 10),
      panel.background = element_rect(fill = "white"), 
      panel.spacing = unit(0,"lines"),
      strip.text.y = element_text(angle = 0) 
    )
  )

  manhplot <- ggplot(data = final_fst_df, 
               aes(x = midpos_Mb, y = fst)) +
    geom_point(fill = PT_COLOR,
               alpha = 0.5, 
               size = 0.1) +
    facet_grid(POP2 ~ chr_num, scales = "free_x") +
    ylab(expression(italic(F[ST]))) +
    xlab("Chromosome position (Mb)") +
    ggtitle(MANHTITLE) +
    scale_y_continuous(breaks = c(0.0, 0.25, 0.5, 0.75, 1)) +
    scale_x_continuous(breaks = c(15,30))
  
  manhplot
  
}

```

## specify
```{r project_setup}

BASEDIR <- "C:/Users/laura.timm/Desktop/Clupea_pallasii/" #the directory where the metadata file lives
PREFIX <- "CPAL-CHAR" #the prefix for the lcWGS run
METADATAFILE <- paste0(BASEDIR, "pacific-herring_metadata.xlsx") #metadatafile name (excel file)
SHEET <- "Sheet1" #sheet containing the pertinent metadata
DEPTH_LOWER_BOUND <- 1
ADMIX_PREFIX_all <- "admixture/CPAL-CHAR_wholegenome-polymorphic"
ADMIX_PREFIX_eBS <- "admixture/CPAL-CHAR-eBS_wholegenome-polymorphic"
ADMIX_PREFIX_GOA <- "admixture/CPAL-CHAR-GOA_wholegenome-polymorphic"

metadata <- read_xlsx(METADATAFILE, sheet = SHEET)

```


# all samples (eBS + GOA)

## Mapping collection locations

```{r map_samples}

specimen_samples <- count(metadata, Latitude_DD, Longitude_DD, MarineRegion)

##with ggmap
#samples_box <- make_bbox(data = specimen_samples, lon = Longitude_DD, lat = Latitude_DD)
#samples_map <- get_map(location = samples_box, source = "google")

#ggmap(samples_map) +
#  geom_jitter(data = specimen_samples, aes(x = lon, y = lat, size = n, color = MarineRegion), alpha = 0.5) +
#  ggtitle("Pacific herring - collection locations") +
#  theme_bw() + 
#  theme(panel.border = element_blank(), panel.grid.major = element_blank())
##########

##with ggOceanMaps
coords <- data.frame(specimen_samples$Latitude_DD)
coords$lon <- specimen_samples$Longitude_DD
colnames(coords) <- c("trans_lat","trans_lon")
trans_coords <- transform_coord(coords)
map_data <- cbind(specimen_samples, trans_coords)

theme_set(
  theme( 
    legend.position = "none",
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, size = 7, vjust = 0.5),
    axis.text.y = element_text(angle = 0, size = 15),
    panel.background = element_rect(fill = "white"), 
    panel.spacing = unit(0,"lines"),
    strip.text.y = element_text(angle = 0) 
  )
)

(map_fig <- basemap(data = map_data, bathymetry = TRUE) +
  geom_jitter(data = map_data, aes(x = trans_lon, y = trans_lat, size = n, color = MarineRegion)) +
  scale_color_manual(values = wes_palette("Darjeeling1", length(unique(metadata$MarineRegion)))))

```

## Aligned sequencing depths

```{r plot_initial_depth_distributions, warning=FALSE, message=FALSE}

just_depths <- read.csv(paste0(BASEDIR, PREFIX, "_depths.csv"), header = FALSE, sep = '\t')
colnames(just_depths) <- c("Barcode", "mean_depth")
full_metadata <- left_join(metadata, just_depths, by = "Barcode")

theme_set(
  theme( 
    panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(),
    axis.text.x = element_text(angle = 90, size = 10, vjust = 0.5),
    axis.text.y = element_text(angle = 0, size = 10),
    panel.background = element_rect(fill = "white"), 
    panel.spacing = unit(0,"lines"),
    strip.text.y = element_text(angle = 0) 
  )
)

(depths_dists_plot <- ggplot(full_metadata, aes(x = mean_depth, color = MarineRegion)) +
  geom_density(n = 50, linewidth = 1) +
  geom_vline(aes(xintercept = DEPTH_LOWER_BOUND), colour = "black") +
  ggtitle("individual mean sequence depth distributions"))

ggsave(paste0(BASEDIR, PREFIX, "-MarineRegion_mean_depths.jpg"), width = 8, height = 5, units = "in", dpi = 300)

```

## PCA

```{r pca_all}

covfile_all <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "_wholegenome-polymorphic.cov")))
features_df_all <- metadata[c("Barcode", "MarineRegion")]
colnames(features_df_all) <- c("Barcode", "feature")

(pca_all <- plot_pca(features_df_all, covfile_all))

```


## Admixture
```{r opt_k_all}

LOGS_all <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_all, "*.log"))
opt_k_fx(LOGS_all, 4)

```



```{r admix_plot_all}

QOPTS_all <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_all, "_*-2.qopt"))
(all_samples_admix_plot <- admix_plot(features_df_all, QOPTS_all))

```


# Manhattan plots

```{r general_genome_scan_setup}

CHROMFILE <- paste0(BASEDIR, "CHAR_chrom_metadata.txt")
chrom_df <- read.table(CHROMFILE, header = TRUE)

```

## east Bering Sea vs Gulf of Alaska
```{r marine-region_manhplot_wholegenome, fig.height = 5, fig.width = 12}

marine_region_POPLIST <- c("eBS", "GOA")
marine_region_FSTFILE <- paste0(BASEDIR, "fst/eBS-GOA.fst.txt")
marine_region_MANHTITLE <- "Pacific herring: east Bering Sea vs Gulf of Alaska"

(marine_region_manhplot <- manhattanplot(chrom_df,
                                        marine_region_POPLIST,
                                        marine_region_FSTFILE,
                                        "blue",
                                        marine_region_MANHTITLE))

```

These...look like different species...I don't like this. Fortunately, there were equal numbers of males and females from each region, so the following manhattan plot shouldn't be impacted by regional differentiation.


## female vs male
```{r sex_manhplot_wholegenome, fig.height = 5, fig.width = 12}

sex_POPLIST <- c("female", "male")
sex_FSTFILE <- paste0(BASEDIR, "fst/female-male.fst.txt")
sex_MANHTITLE <- "Pacific herring: male vs female"

(sex_manhplot <- manhattanplot(chrom_df,
                              sex_POPLIST,
                              sex_FSTFILE,
                              "red",
                              sex_MANHTITLE))

```


Really clear signal on chr8 (which is also where the sex-determining gene is found in Atlantic herring - hooray!)

Let's start considering these two clusters (eBS and GOA as separate groups.)

# east Bering Sea

## PCA

### N = 20
```{r pca_eBS}

eBS_metadata <- full_metadata[full_metadata$MarineRegion == "eBS",]
features_df_eBS <- eBS_metadata[c("Barcode", "Sex")]
colnames(features_df_eBS) <- c("Barcode", "feature")

covfile_eBS <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-eBS_wholegenome-polymorphic.cov")))

(pca_eBS <- plot_pca(features_df_eBS, covfile_eBS))

```

Maybe two or three groups? Remove outliers to see it better.

### N = 18

```{r pca_eBS_18}

#identify which individuals to filter out
pca_e_eBS <- eigen(covfile_eBS)
first3pcs_eBS <- data.frame(pca_e_eBS$vectors[,1:3])
labeled_first3pcs_eBS <- cbind(features_df_eBS,
                             pc1 = first3pcs_eBS$X1,
                             pc2 = first3pcs_eBS$X2,
                             pc3 = first3pcs_eBS$X3)
(filter_eBS_18 <- labeled_first3pcs_eBS[labeled_first3pcs_eBS$pc1 > 0.4,])

eBS_metadata_18 <- labeled_first3pcs_eBS[labeled_first3pcs_eBS$pc1 < 0.4,]
features_df_eBS_18 <- eBS_metadata_18[c("Barcode", "feature")]

covfile_eBS_18 <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-eBS_wholegenome-polymorphic_keep18.cov")))

(pca_eBS_18 <- plot_pca(features_df_eBS_18, covfile_eBS_18))

```

Yeah, two groups in the eBS sample. Which is a little weird...I wonder if more detailed collection information exists for these individuals. Unfortunately, sample sizes are getting a little too small to genome scan FST confidnetly (I mean, I'm probably still gonna...).
Let's check for signal on chromosome-level PCAs!

### N = 18, by-chromosome
```{r pca_by_chrom_eBS_18}

for(i in 1:length(unique(chrom_df$chr_num))){
  plotChrom_eBS <- unique(chrom_df$chr_num)[i]
  chrom_cov_eBS <- as.matrix(read.table(paste0(BASEDIR, 
                                     "pca/", 
                                     PREFIX, 
                                     "-eBS_", 
                                     unique(chrom_df$chr)[i], 
                                     "_polymorphic.cov")))
  
  (pca_by_chrom_eBS_18 <- plot_pca(features_df_eBS_18, chrom_cov_eBS))
  ggsave(paste0(BASEDIR, "pca/", PREFIX, "-eBS18_pca_", plotChrom_eBS, ".jpg"), width = 8, height = 6, units = "in")
}

```

Chromosomes that look like they have signal (without more detailed collection data, hard to know if it's adaptive):
chr2
chr6
chr7
chr12
chr13
chr16
chr17
chr20
chr23


## admixture
```{r opt_k_eBS}

LOGS_eBS <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_eBS, "*.log"))
opt_k_fx(LOGS_eBS, 5)

```



```{r admix_plot_eBS}

QOPTS_eBS <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_eBS, "_*-2.qopt"))
(eBS_samples_admix_plot <- admix_plot(features_df_eBS, QOPTS_eBS))

```

## manhattan plots
```{r manhplot_sex_eBS}

eBS_sex_POPLIST <- c("eBS_female", "eBS_male")
eBS_sex_FSTFILE <- paste0(BASEDIR, "fst/eBS_female-eBS_male.fst.txt")
eBS_sex_MANHTITLE <- "Pacific herring, eBS: male vs female"

(eBS_sex_manhplot <- manhattanplot(chrom_df,
                              eBS_sex_POPLIST,
                              eBS_sex_FSTFILE,
                              "red",
                              eBS_sex_MANHTITLE))

```


# Gulf of Alaska

## PCA

### N = 20
```{r pca_GOA}

GOA_metadata <- full_metadata[full_metadata$MarineRegion == "GOA",]
features_df_GOA <- GOA_metadata[c("Barcode", "Sex")]
colnames(features_df_GOA) <- c("Barcode", "feature")

covfile_GOA <- as.matrix(read.table(paste0(BASEDIR, "pca/", PREFIX, "-GOA_wholegenome-polymorphic.cov")))

(pca_GOA <- plot_pca(features_df_GOA, covfile_GOA))

```

Looks like two groups here as well, but sample sizes are getting too small to be sure. Like eBS, let's look at chromosome-level PCAs.

### by-chromosome
```{r pca_by_chrom_GOA}

for(i in 1:length(unique(chrom_df$chr_num))){
  plotChrom_GOA <- unique(chrom_df$chr_num)[i]
  chrom_cov_GOA <- as.matrix(read.table(paste0(BASEDIR, 
                                     "pca/", 
                                     PREFIX, 
                                     "-GOA_", 
                                     unique(chrom_df$chr)[i], 
                                     "_polymorphic.cov")))
  
  (pca_by_chrom_GOA <- plot_pca(features_df_GOA, chrom_cov_GOA))
  ggsave(paste0(BASEDIR, "pca/", PREFIX, "-GOA_pca_", plotChrom_GOA, ".jpg"), width = 8, height = 6, units = "in")
}

```

Chromosomes that look like they have signal (without more detailed collection data, hard to know if it's adaptive):
chr6
chr7
chr8 (not sex, three clusters)
chr12
chr15
chr16

## admixture
```{r opt_k_GOA}

LOGS_GOA <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_GOA, "*.log"))
opt_k_fx(LOGS_GOA, 5)

```

```{r admix_GOA}

QOPTS_GOA <- Sys.glob(paste0(BASEDIR, ADMIX_PREFIX_GOA, "_*-2.qopt"))
(GOA_samples_admix_plot <- admix_plot(features_df_GOA, QOPTS_GOA))

```

## manhattan plots
```{r manhplot_sex_GOA}

GOA_sex_POPLIST <- c("GOA_female", "GOA_male")
GOA_sex_FSTFILE <- paste0(BASEDIR, "fst/GOA_female-GOA_male.fst.txt")
GOAsex_MANHTITLE <- "Pacific herring, Gulf of Alaska: male vs female"

(GOA_sex_manhplot <- manhattanplot(chrom_df,
                              GOA_sex_POPLIST,
                              GOA_sex_FSTFILE,
                              "red",
                              GOAsex_MANHTITLE))

```
